File: frontend/dreamhome-angular/src/app/interfaces/lease-contract.ts
import { MonoTypeOperatorFunction } from "rxjs";

export interface LeaseContract {
    contractId: Number;
    leaseAmount: Number;
    dateOfSignature: String;
    additionalCosts: Number;
    numberOfPersons: Number;
    renterId: Number;
    apartmentId: Number;
    active: String;
}

File: backend/property_management_system/src/main/java/de/hs_esslingen/property_management_system/entities/LeaseContract.java
package de.hs_esslingen.property_management_system.entities;

import java.time.LocalDate;

public class LeaseContract {

    public Integer contractId;
    public Double leaseAmount;
    public LocalDate dateOfSignature;
    public Double additionalCosts;
    public Integer numberOfPersons;
    public Integer renterId;
    public Integer apartmentId;
    public Character active;

    public LeaseContract() {
    }
    
}
